### PR TITLE
Speedup

### PR DESCRIPTION
**Pull request by MajorCaiger**
Re-submitted after resolving conflicts as requested.

### CODE REVIEW

File: src/Driver/Driver.php [hunk 1]
```diff
@@ -39,7 +39,7 @@
     /**
      * Start collection of code coverage information.
      */
-    public function start();
+    public function start($determineUnusedAndDead = true);
```
Comment (GrahamCampbell) [inline src/Driver/Driver.php]: you forgot to update the phpdoc

**Comment by GrahamCampbell** [comment]
This is a major breaking change due to the modification of an interface. Existing implementations will cause php to fatal error.

**Comment by GrahamCampbell** [comment]
Oh, I see the master is 4.x. That's fine then. ;)

File: src/Driver/Driver.php (hunk 1 above)
Comment (MajorCaiger) [inline src/Driver/Driver.php]: Good point, I've added these now

**Comment by sebastianbergmann** [comment]
Thanks, @MajorCaiger. Did you do a benchmark and can share some numbers?

**Comment by MajorCaiger** [comment]
The original benchmarks can be found here https://github.com/sebastianbergmann/php-code-coverage/pull/384 I think the implementation changed slightly but the improvements should be similar.

**Comment by liorharbater** [comment]
Hi @sebastianbergmann 
Using this Speedup enhancement would greatly assist us in reducing our coverage tests cycle time!
Do you plan to release it as part of an official phpunit 4.X release any time soon?
We are using php 5.3 and upgrading to any, more advanced, php version will be too complex at this stage, so we are limited to the 4.X phpunit.

Thanks,
Lior

**Comment by sebastianbergmann** [comment]
There will be no major changes to anything related to PHPUnit 4.8, sorry.

**Comment by liorharbater** [comment]
Unfortunate, but understandable :)
Any plans to release this Speedup enhancement as part of PHPUnit 5.X, any time soon ?

**Comment by GrahamCampbell** [comment]
This is already in PHPUnit 5.x is it not?

**Comment by willemstuursma** [comment]
We've seen a speed increase from 1h48m to 13m. I just want to say I really appreciate this change.

**Comment by MajorCaiger** [comment]
@willemstuursma That's a fantastic improvement!

**Pull request merged**